kustomization.yaml

```diff
@@ -0,0 +1,3 @@
+resources:
```

Comment:
with this change we'll have kustomization files living in 3 places - `root`, `internal/provider/kubernetes/` and `internal/infrastructure/kubernetes`, can we simplify this ?

Reply:
I'm concerned about making too many changes during the RC2 release. I have confirmed that this PR works so I'd like to proceed with it and finish the release. I created https://github.com/envoyproxy/gateway/issues/479 to discuss how to improve the current kustomization setup.